File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/achievement/AchievementDetailedListRequestMessage.scala





















// Generated on 05/03/2013 10:34:32
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait AchievementDetailedListRequestMessage extends MessageDeserializer {
  val messageId = 6357

  protected trait Message extends BaseMessage with Serializable {
    val messageId = AchievementDetailedListRequestMessage.messageId

    val categoryId: Short

    def serialize(buf: Buffer) {
      buf.writeShort(categoryId)
    }

  }

  protected def create(categoryId: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val categoryId = buf.readShort()

    create(categoryId)
  }

}

object AchievementDetailedListRequestMessage extends AchievementDetailedListRequestMessage {
  case class MessageImpl(categoryId: Short) extends Message
  type Target = MessageImpl

  override def create(categoryId: Short) = MessageImpl(categoryId)

  def apply(categoryId: Short) = create(categoryId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/AchievementDetailedListRequestMessage.java


// Generated on 05/08/2013 19:37:38
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class AchievementDetailedListRequestMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6357;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short categoryId;
    
    public AchievementDetailedListRequestMessage() { }
    
    public AchievementDetailedListRequestMessage(short categoryId) {
        this.categoryId = categoryId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(categoryId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        categoryId = buf.readShort();
        if (categoryId < 0)
            throw new RuntimeException("Forbidden value on categoryId = " + categoryId + ", it doesn't respect the following condition : categoryId < 0");
    }
    
}
